lib/internal/Magento/Framework/App/Response/HttpInterface.php

```diff
@@ -16,4 +16,49 @@
      * @return void
      */
     public function setHttpResponseCode($code);
+
```

Comment:
I don't mind merging. 
To clarify what I initially meant, here is one edge case which might be broken: extension developer creates implementation of `HttpInterface`, and uses it only in his own module (does not set as global preference, but as a dependency for specific class using `type` declaration in `di.xml`).
In this case implementation might have only 2 methods and will be broken after our changes.